File: scripts/check_football_data.mjs
import { footballAllSeasons, footballSeasonPlayersMap } from '../src/data/footballPlayers.js';

console.log('Total seasons:', footballAllSeasons.length);
console.log('Seasons with < 5 players:');
for (const season of footballAllSeasons) {
  const players = footballSeasonPlayersMap[season];
  if (players.length < 5) {
    console.log(`  ${season}: ${players.length} players`);
  }
}

console.log('---');
console.log('Seasons with no QBs:');
for (const season of footballAllSeasons) {
  const players = footballSeasonPlayersMap[season];
  const qbs = players.filter(p => p.primaryPosition === 'QB');
  if (qbs.length === 0) console.log(`  ${season}`);
}

console.log('---');
console.log('Seasons with no RBs:');
for (const season of footballAllSeasons) {
  const players = footballSeasonPlayersMap[season];
  const rbs = players.filter(p => p.primaryPosition === 'RB');
  if (rbs.length === 0) console.log(`  ${season}`);
}

console.log('---');
console.log('Seasons with no WRs:');
for (const season of footballAllSeasons) {
  const players = footballSeasonPlayersMap[season];
  const wrs = players.filter(p => p.primaryPosition === 'WR');
  if (wrs.length === 0) console.log(`  ${season}`);
}

console.log('---');
console.log('Seasons with no TEs:');
for (const season of footballAllSeasons) {
  const players = footballSeasonPlayersMap[season];
  const tes = players.filter(p => p.primaryPosition === 'TE');
  if (tes.length === 0) console.log(`  ${season}`);
}

console.log('---');
console.log('Sample of season sizes:');
for (const season of footballAllSeasons.slice(0, 5)) {
  const players = footballSeasonPlayersMap[season];
  console.log(`  ${season}: ${players.length} players`);
}
for (const season of footballAllSeasons.slice(-5)) {
  const players = footballSeasonPlayersMap[season];
  console.log(`  ${season}: ${players.length} players`);
}

File: scripts/check_football_data.py
import sys
sys.path.insert(0, '/Users/danielsmac/Documents/Kentucky Basketball Team Draft/src')

from data.footballPlayers import footballAllSeasons, footballSeasonPlayersMap

print('Total seasons:', len(footballAllSeasons))
print('Seasons with < 5 players:')
for season in footballAllSeasons:
    players = footballSeasonPlayersMap[season]
    if len(players) < 5:
        print(f'  {season}: {len(players)} players')

print('---')
print('Seasons with no QBs:')
for season in footballAllSeasons:
    players = footballSeasonPlayersMap[season]
    qbs = [p for p in players if p['primaryPosition'] == 'QB']
    if len(qbs) == 0:
        print(f'  {season}')

print('---')
print('Seasons with no RBs:')
for season in footballAllSeasons:
    players = footballSeasonPlayersMap[season]
    rbs = [p for p in players if p['primaryPosition'] == 'RB']
    if len(rbs) == 0:
        print(f'  {season}')

print('---')
print('Seasons with no WRs:')
for season in footballAllSeasons:
    players = footballSeasonPlayersMap[season]
    wrs = [p for p in players if p['primaryPosition'] == 'WR']
    if len(wrs) == 0:
        print(f'  {season}')

print('---')
print('Seasons with no TEs:')
for season in footballAllSeasons:
    players = footballSeasonPlayersMap[season]
    tes = [p for p in players if p['primaryPosition'] == 'TE']
    if len(tes) == 0:
        print(f'  {season}')
